registration/registration-client/src/main/java/io/mosip/registration/controller/reg/DemographicDetailController.java

```diff
@@ -2372,15 +2385,40 @@ private void retrieveAndPopulateLocationByHierarchy(ComboBox<LocationDto> srcLoc
 				"Retrieving and populating of location by selected hirerachy started");
 
 		try {
-			LocationDto selectedLocationHierarchy = srcLocationHierarchy.getSelectionModel().getSelectedItem();
-			destLocationHierarchy.getItems().clear();
-			destLocationHierarchyInLocal.getItems().clear();
 
+			LocationDto selectedLocationHierarchy = srcLocationHierarchy.getSelectionModel().getSelectedItem();
 			if (selectedLocationHierarchy != null) {
-				destLocationHierarchy.getItems().addAll(masterSync.findProvianceByHierarchyCode(
-						selectedLocationHierarchy.getCode(), selectedLocationHierarchy.getLangCode()));
-				destLocationHierarchyInLocal.getItems().addAll(masterSync.findProvianceByHierarchyCode(
-						selectedLocationHierarchy.getCode(), ApplicationContext.localLanguage()));
+				destLocationHierarchy.getItems().clear();
+				destLocationHierarchyInLocal.getItems().clear();
+
+				if (selectedLocationHierarchy.getCode().equalsIgnoreCase("NA")) {
+					destLocationHierarchy.getItems().add(selectedLocationHierarchy);
+					destLocationHierarchyInLocal.getItems().add(selectedLocationHierarchy);
+				} else {
+					List<LocationDto> locations = masterSync.findLocationByHierarchyCode(
+							selectedLocationHierarchy.getCode(), selectedLocationHierarchy.getLangCode());
+
+					List<LocationDto> locationsSecondary = masterSync.findProvianceByHierarchyCode(
+							selectedLocationHierarchy.getCode(), ApplicationContext.localLanguage());
+
+					if (locations.isEmpty()) {
+						LocationDto lC = new LocationDto();
+						lC.setCode("NA");

```

Comment:
dont hard code "NA" instead create a constant file for all UI constants. It will cause sonar violation otherwise